src/wp-includes/script-loader.php

```diff
@@ -1843,6 +1843,50 @@ function wp_just_in_time_script_localization() {
 	);
 }
 
+/**
+ * Prints a loader script if there is text/template registered script.
+ *
+ * @since 6.3.0
+ *
+ * When added to the DOM, this script converts any text/template scripts
+ * associated with a handle into type/javascript, and executes them.
+ */
+function wp_print_delayed_inline_script_loader() {
+	$wp_scripts = wp_scripts();
+
+	if ( $wp_scripts->has_delayed_inline_script() ) {
+		$output    = <<<JS
+(function () {
+  // Capture the nonce of the currentScript so we can use it when evaluating after inline scripts.
+  var nonce = document.currentScript.nonce;
+
+  window.wpLoadAfterScripts = function wpLoadAfterScripts(handle) {
+    var scripts, newScript, i, len;
+    scripts = document.querySelectorAll(
+      '[type="text/template"][data-wp-executes-after="' + handle + '"]'
+    );
+    for (i = 0, len = scripts.length; i < len; i++) {
+      if (nonce && nonce !== scripts[i].nonce) {
+        console.error(
+          "CSP nonce check failed for after inline script. Execution aborted.",
+          scripts[i]
+        );
+        continue;
+      }
+      newScript = scripts[i].cloneNode(true);
+      newScript.type = "text/javascript";
+      scripts[i].parentNode.replaceChild(newScript, scripts[i]);
+    }
+  };
+})();
+JS;
+		$type_attr         = current_theme_supports( 'html5', 'script' ) ? '' : 'text/javascript';
+		$script_type_array = '' === $type_attr ? array() : array( 'type' => $type_attr );
+
+		wp_print_inline_script_tag( $output, array_merge( $script_type_array, array( 'id' => 'wp-executes-after-js' ) ) );
```

Comment:
It turns out that `wp_get_inline_script_tag()` actually automatically handles adding the `type` attribute if not `html5`. So this can be simplified:

```suggestion
		wp_print_inline_script_tag( $output, array( 'id' => 'wp-executes-after-js' ) );
```

Reply:
Nice catch @westonruter, this has been resolved in a recent commit thank you.